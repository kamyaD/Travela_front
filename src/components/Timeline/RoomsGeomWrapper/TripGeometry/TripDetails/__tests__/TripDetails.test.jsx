import React from 'react';
import TimelineDetails from '..';

const props = {
  trip: {
    checkInDate: '2018-01-01',
    checkOutDate: '2018-01-10',
    request: {
      name: 'Alice Doe'
    }
  },
  bookingDetailsPos: '0px',
  detailsVariantClass: '',
  toggleBookingDetails: jest.fn(),
  translateDetailsLeft: false,
  handleChangeRoomModal: jest.fn()
};

describe('<TimelineDetails />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<TimelineDetails {...props} />);
  });

  it('renders prorperly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders properly without a checkInDate or checkOutDate', () => {
    props.trip.checkInDate = null;
    props.trip.checkOutDate = null;
    wrapper = shallow(<TimelineDetails {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('closes on click close button, calls the toggler with right arg(s)', () => {
    const closeButton = wrapper.find('.close-button');
    const onClickCloseListener = closeButton.prop('onClick');
    onClickCloseListener();
    expect(props.toggleBookingDetails).toHaveBeenCalledWith('close');
  });

  it('it does not close when other areas of the trip details besides the \
  close button are clicked', () => {
    const detailsContainer = wrapper.find('.trip-booking-details');
    const clickDetailsListener = detailsContainer.prop('onClick');
    const e = {stopPropagation: jest.fn()};
    clickDetailsListener(e);
    expect(e.stopPropagation).toHaveBeenCalledTimes(1);
  });

  it('calls the handleChangeRoomModal function onclick', () => {
    wrapper = shallow(<TimelineDetails {...props} />);
    const submitButton = wrapper.find('button');
    submitButton.simulate('click');
    expect(props.handleChangeRoomModal).toHaveBeenCalled();
  })
});
