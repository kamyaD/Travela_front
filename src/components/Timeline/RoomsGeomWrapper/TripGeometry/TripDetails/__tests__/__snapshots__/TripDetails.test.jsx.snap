// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimelineDetails /> renders properly without a checkInDate or checkOutDate 1`] = `
<div
  className="trip-booking-details details- "
  onClick={[Function]}
  role="presentation"
  style={
    Object {
      "left": "0pxpx",
      "position": "absolute",
    }
  }
>
  <div
    className="trip-booking-details__header"
  >
    <span>
      Alice Doe
    </span>
    <div
      className="close-button"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="presentation"
      tabIndex="-1"
    >
      <div>
        ×
      </div>
    </div>
  </div>
  <div
    className="trip-booking-details__body"
  >
    <div>
      <span>
        Status
      </span>
      <span>
        Booked
      </span>
    </div>
    <div>
      <span>
        Check-in
      </span>
      <span>
        n/a
      </span>
    </div>
    <div>
      <span>
        Check-out
      </span>
      <span>
        n/a
      </span>
    </div>
    <div>
      <span>
        Duration
      </span>
      <span>
        n/a
      </span>
    </div>
    <div>
      <span>
        Origin
      </span>
      <span />
    </div>
    <div>
      <span>
        <button
          className="trip-booking-details__button"
          onClick={[Function]}
          type="button"
        >
          Change Room
        </button>
      </span>
    </div>
  </div>
</div>
`;

exports[`<TimelineDetails /> renders prorperly 1`] = `
<div
  className="trip-booking-details details- "
  onClick={[Function]}
  role="presentation"
  style={
    Object {
      "left": "0pxpx",
      "position": "absolute",
    }
  }
>
  <div
    className="trip-booking-details__header"
  >
    <span>
      Alice Doe
    </span>
    <div
      className="close-button"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="presentation"
      tabIndex="-1"
    >
      <div>
        ×
      </div>
    </div>
  </div>
  <div
    className="trip-booking-details__body"
  >
    <div>
      <span>
        Status
      </span>
      <span>
        Booked
      </span>
    </div>
    <div>
      <span>
        Check-in
      </span>
      <span>
        01 Jan 2018
      </span>
    </div>
    <div>
      <span>
        Check-out
      </span>
      <span>
        10 Jan 2018
      </span>
    </div>
    <div>
      <span>
        Duration
      </span>
      <span>
          1 Week 3 Days
      </span>
    </div>
    <div>
      <span>
        Origin
      </span>
      <span />
    </div>
    <div>
      <span>
        <button
          className="trip-booking-details__button"
          onClick={[Function]}
          type="button"
        >
          Change Room
        </button>
      </span>
    </div>
  </div>
</div>
`;
